qcodes/tests/parameter/test_delegate_parameter.py

```diff
@@ -512,3 +513,27 @@ def test_underlying_instrument_property_for_delegate_parameter():
 
     d = DelegateParameter('delegate_parameter_without_source', source=None)
     assert d.underlying_instrument is None
+
+def test_value_validation():
+    source_param = Parameter("source", set_cmd=None, get_cmd=None)
+    delegate_param = DelegateParameter("delegate", source=source_param)
+
+    delegate_param.vals = vals.Numbers(-10, 10)
+    source_param.vals = None
+    delegate_param.validate(1)
+    with pytest.raises(ValueError):
+        delegate_param.validate(11)
+
+    delegate_param.vals = None
+    source_param.vals = vals.Numbers(-5, 5)
+    delegate_param.validate(1)
+    with pytest.raises(ValueError):
+        delegate_param.validate(6)
+
+    delegate_param.vals = vals.Numbers(-10, 10)
+    source_param.vals = vals.Numbers(-5, 5)
+    delegate_param.validate(1)
+    with pytest.raises(ValueError):
+        delegate_param.validate(6)
+    with pytest.raises(ValueError):
+        delegate_param.validate(11)
```

Comment:
I am not sure how this would interact with having a scale and or offset on the delegate parameter. It would probably be nice to add tests to ensure that works correctly

Reply:
I have just checked that .validate only checks on the value and ignores the raw_value (where offset and scale are considered).
I will add a couple more tests in the next commit.

On the other hand, while checking the effect of offset and scale, I found out that DelegateParameter does not inherit the offset and scale from the source parameter (see the snippet below). Is this intentional?

```python
param = Parameter('param', set_cmd=None, get_cmd=None, offset=10)
delegated = DelegateParameter('delegated', source=param) # no offset defined

print(f'{param.offset=}') # returns 10
print(f'{delegated.offset=}') # returns None

param(0) 
print(f'{param.get()=}') # returns 0
print(f'{param.get_raw()=}') # returns 10 as expected

delegated(0)
print(f'{delegated.get()=}') # returns 0
print(f'{delegated.get_raw()=}') # returns 0 --> it ignores source offset.
```

Reply:
Yes, I belive the idea is that you can have a delegate parameter in a different unit. 
So you could do something like having a temperate in Kelvin with a delegate parameter in Celsius or Fahrenheit

Reply:
ah ok, that makes sense, thanks!